serverless-plugin/alarms-sqs.js

```diff
@@ -0,0 +1,132 @@
+'use strict'
+
+/**
+ * @param {object} sqsAlarmsConfig The fully resolved alarm configuration
+ */
+module.exports = function sqsAlarms (sqsAlarmsConfig, context) {
+  return {
+    createSQSAlarms
+  }
+
+  /**
+   * Add all required SQS alarms to the provided CloudFormation template
+   * based on the SQS resources found within
+   *
+   * @param {CloudFormationTemplate} cfTemplate A CloudFormation template object
+   */
+  function createSQSAlarms (cfTemplate) {
+    const queueResources = cfTemplate.getResourcesByType(
+      'AWS::SQS::Queue'
+    )
+
+    for (const [queueResourceName, queueResource] of Object.entries(
+      queueResources
+    )) {
+      if (sqsAlarmsConfig.InFlightMessagesPc.enabled) {
+        const inFlightMsgsAlarm = createInFlightMsgsAlarm(
+          queueResourceName,
+          queueResource,
+          sqsAlarmsConfig.ApproximateAgeOfOldestMessage
+        )
+        cfTemplate.addResource(inFlightMsgsAlarm.resourceName, inFlightMsgsAlarm.resource)
+      }
+
+      if (sqsAlarmsConfig.AgeOfOldestMessage.enabled) {
+        if (sqsAlarmsConfig.AgeOfOldestMessage.Threshold == null) {
+          throw new Error('SQS AgeOfOldestMessage alarm is enabled but `Threshold` is not specified. Please specify a threshold or disable the alarm.')
+        }
+
+        const oldestMsgAgeAlarm = createOldestMsgAgeAlarm(
+          queueResourceName,
+          queueResource,
+          sqsAlarmsConfig.Invocations
+        )
+        cfTemplate.addResource(
+          oldestMsgAgeAlarm.resourceName,
+          oldestMsgAgeAlarm.resource
+        )
+      }
+    }
+  }
+
+  function createSqsAlarm (
+    alarmName,
+    alarmDescription,
+    queueName,
+    comparisonOperator,
+    threshold,
+    metrics,
+    metricName,
+    statistic,
+    period
+  ) {
+    const metricProperties = metrics
+      ? { Metrics: metrics }
+      : {
+          Dimensions: [{ Name: 'QueueName', Value: queueName }],
+          MetricName: metricName,
+          Namespace: 'AWS/SQS',
+          Period: period,
+          Statistic: statistic
+        }
+
+    return {
+      Type: 'AWS::CloudWatch::Alarm',
+      Properties: {
+        ActionsEnabled: true,
+        AlarmActions: [context.topicArn],
+        AlarmName: alarmName,
+        AlarmDescription: alarmDescription,
+        EvaluationPeriods: 1,
```

Comment:
I think we are missing loading of `EvaluationPeriods` and `TreatMissingData` from the config. I only noticed this now and it's for all alarms!